src/restaurants/restaurants.service.ts

```diff
@@ -1,8 +1,41 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Restaurant } from './entities/restaurants.entity';
+import { CreateRestaurantDto } from './dto/create-restaurants.dto';
+import { UpdateRestaurantDto } from './dto/update-restaurants.dto';
 
 @Injectable()
 export class RestaurantsService {
-  getRestaurants(): string {
-    return 'RestaurantsService 111';
+  constructor(
+    @InjectRepository(Restaurant)
+    private readonly restaurantRepository: Repository<Restaurant>,
+  ) {}
+
+  async getRestaurantById(restaurantId: string): Promise<Restaurant> {
+    const restaurant = await this.restaurantRepository.findOne({ where: { id: restaurantId } });
+    if (!restaurant) {
+      throw new NotFoundException(`Restaurant with ID ${restaurantId} not found`);
+    }
+    return restaurant;
+  }
+
+  async createRestaurant(userId: string, createRestaurantDto: CreateRestaurantDto): Promise<Restaurant> {
+    const newRestaurant = this.restaurantRepository.create({
+      ...createRestaurantDto,
+      user_id: userId,
+    });
+    return await this.restaurantRepository.save(newRestaurant);
+  }
+
+  async updateRestaurant(restaurantId: string, updateRestaurantDto: UpdateRestaurantDto): Promise<Restaurant> {
+    const restaurant = await this.restaurantRepository.findOne({ where: { id: restaurantId } });
+
+    if (!restaurant) {
+      throw new NotFoundException(`Restaurant with ID ${restaurantId} not found`);
+    }
+
+    Object.assign(restaurant, updateRestaurantDto);
+    return await this.restaurantRepository.save(restaurant);
```

Comment:
이렇게 기존 객체를 변경하는 방식으로 코드를 작성하기 보다는 변경된 값을 받아서 이후 처리를 하는 방식으로 코드를 작성하는게 더 좋을 것 같습니다~
```suggestion
    const newRestaurant = Object.assign(restaurant, updateRestaurantDto);
    return await this.restaurantRepository.save(newRestaurant);
```

```suggestion
    const newRestaurant = { ...restaurant, ...updateRestaurantDto }
    return await this.restaurantRepository.save(restaurant);
```